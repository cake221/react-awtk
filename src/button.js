function applicationInit() {
  var win = TWindow.create(null, 0, 0, 0, 0);
  var ok = TButton.create(win, 0, 0, 0, 0);

  ok.setText("ok");
  ok.setSelfLayoutParams("center", "middle", "50%", "30");

  ok.on(TEventType.CLICK, function(evt) {
    var e = TPointerEvent.cast(evt);
    console.log("global: " + TGlobal.getPointerX() + " " + TGlobal.getPointerY() + " " + TGlobal.isPointerPressed());
    console.log("on click: " + e.x + " " + e.y);

    return TRet.OK;
  });
  
  console.log("global: " + TGlobal.getPointerX() + " " + TGlobal.getPointerY() + " " + TGlobal.isPointerPressed());

  win.layout();
}

export {
  applicationInit
}

