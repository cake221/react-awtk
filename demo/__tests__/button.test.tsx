import "../button"

describe("test button", ()=>{

  it("toMatchSnapshot", ()=>{
    expect(__AwtkSnapshot).toMatchSnapshot()
  })
})
