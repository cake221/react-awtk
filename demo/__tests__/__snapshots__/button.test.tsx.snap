// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test button toMatchSnapshot 1`] = `
Array [
  "getRootHostContext",
  "getChildHostContext",
  "shouldSetTextContent",
  "getChildHostContext",
  "shouldSetTextContent",
  "createInstance",
  "funtion  button_create( parent, x, y, w, h ) :null, 0, 0, 0, 0",
  "funtion  widget_set_self_layout_params( widget, x, y, w, h ) :undefined, center, middle, 50%, 30",
  "funtion  widget_set_text_utf8( widget, text ) :undefined, ok",
  "funtion  widget_on( widget, type, on_event, ctx ) :undefined, EVT_CLICK, function (evt) {
                    var e = src_1.TPointerEvent.cast(evt);
                    var target = src_1.TButton.cast(e.target);
                    console.log(\\"click at \\" + e.x + \\" \\" + e.y + \\" on \\" + target.name);
                    console.log(\\"global: \\" + src_1.TGlobal.getPointerX() + \\" \\" + src_1.TGlobal.getPointerY() + \\" \\" + src_1.TGlobal.isPointerPressed());
                    return TRet.OK;
                }, null",
  "finalizeInitialChildren",
  "createInstance",
  "funtion  window_create( parent, x, y, w, h ) :null, 0, 0, 0, 0",
  "appendInitialChild",
  "finalizeInitialChildren",
  "prepareForCommit",
  "appendNodeChild",
  "resetAfterCommit",
  "getPublicInstance",
  "funtion  widget_add_child( widget, child ) :undefined, [object Object]",
  "funtion  widget_layout( widget ) :undefined",
  "win1 子控件 数量: 1",
  "funtion  tk_get_pointer_x(  ) :",
  "funtion  tk_get_pointer_y(  ) :",
  "funtion  tk_is_pointer_pressed(  ) :",
]
`;
